### DISCUSSION TITLE
Unable to Clone a private repository using Personal access token and Fine grained token in local using command Prompt

### DISCUSSION THREAD

**Discussion by JVerma-gls** [community/community#144342] [Actions]
### Select Topic Area

General

### Body

I am unable to Clone a private repository using Personal access token and Fine grained token in local using command Prompt. I am running following command in cmd. My goal is to integrate Git repo with snowflake, I was able to create SECRET and API INTEGRATION objects in snowflake, however when I tried creating GIT REPOSITORY, I was getting this error, "Failed to access the Git Repository. Operation 'clone' is not authorized."

Then I tried cloning my private repository locally as follows and got error message as follows.

https://<TOKEN>@github.com/<username>/<repository>.git

Here is the error message I got.

remote: Personal access tokens (classic) are forbidden from accessing this repository.
fatal: unable to access 'https://github.com/username/Snowflaketest/': The requested URL returned error: 403

When I tried using fine grained token, I got the following error.

remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/username/Snowflaketest/': The requested URL returned error: 403

Let me know how can I resolve this issue or what might be causing this.

**Comment by M-tech-cmd** [comment]
1. Check Your Token Permissions:

Ensure that your personal access token (PAT) has the necessary permissions. For cloning, you typically need repo scope for classic tokens or the appropriate repository permissions for fine-grained tokens.

2. Correct URL Format:
Make sure your clone URL includes your username. The format should be:
git clone https://<username>:<token>@github.com/<username>/<repository>.git

3. Use SSH Instead of HTTPS:
If HTTPS continues to cause issues, consider using SSH. First, set up SSH keys on GitHub, then use the SSH URL:
git clone git@github.com:<username>/<repository>.git

4. Update Git:
Ensure you are using the latest version of Git. Older versions might have issues with newer authentication methods.

5. Check Repository Access:
Verify that your GitHub account has the necessary access to the repository. If it’s an organization repository, ensure your token is authorized to access it.

6.Credential Caching:
If you are using credential caching, make sure the cached credentials are up-to-date. You can clear the cache and re-enter your credentials.

**Reply by PatrickCaneloDigital** [reply]
I can add to this, because I had a similar same issue.
Yesterday I somehow was able to clone my private repo with a fine grained PAT. But today I wanted to make a pull request and git asked me for a password, but then denied access because username/password access was removed.

I read in another place that PAT in that use seem to work only for one time access.
My webhosting does not allow ssh-add, so ssh is out of question.

I know it's another person's issue, but I think there are "sinergies".
When I clone with https://MyTokenName:MyToken@github.com/MyAccountName/myrepo.git

I get this
```
git clone https://MyTokenName:MyToken@github.com/MyAccountName/myrepo.git
Cloning into 'myrepo'...
remote: Support for password authentication was removed on August 13, 2021.

```

Any combination using only the token gives the same error, at latest in the second access:
```
git clone https://${GITHUB_TOKEN}@github.com/MyAccountName/myrepo.git
Cloning into 'myrepo'...
Password for 'https://github_mytoken@github.com': 
remote: Support for password authentication was removed on August 13, 2021.

```

My actual conclusion is that PAT in that form is simply not useable...
The ssh method would be if I'd be able to use ssh-add on my hosting

**Comment by Gameonn** [comment]
I am facing the same issue. Even after adding **Personal Access Token(classic)** still it is not working out.

I am getting another error after this

> fatal: unable to access 'https://github.com:/repo-web.git/': URL rejected: Port number was not a decimal number between 0 and 65535

**Comment by richenyadav001** [comment]
Sometime personal access token has special character which needs to to encoded 
Run in any browsers console, you will get encoded access token 
encodeURIComponent("your_access_token")

**Comment by singh-adarsh1** [comment]
In my case I was using a fine grained personal token which couldn't access the org's repos, a token (classic) did the job

**Comment by ruseel** [comment]
I got same issue. I think that "I created fine-grained-access-token right" but It didn't worked. 
but after I re-create fine-grained personal access token (pat) with "Contents" permission, now I can clone. 

with url 
  http://<my_id>:<token>@github.com/<my_id>/<repo>.git 
  
So create pat with "Contents" permission for fine-grained pat.
